cmd/kube-controller-manager/app/certificates.go

```diff
@@ -89,22 +89,45 @@ func startCSRSigningController(ctx ControllerContext) (http.Handler, bool, error
 	}
 
 	c := ctx.ClientBuilder.ClientOrDie("certificate-controller")
+	csrInformer := ctx.InformerFactory.Certificates().V1beta1().CertificateSigningRequests()
+	certTTL := ctx.ComponentConfig.CSRSigningController.ClusterSigningDuration.Duration
+	caFile, caKeyFile := getKubeletServingSignerFiles(ctx.ComponentConfig.CSRSigningController)
 
-	signer, err := signer.NewCSRSigningController(
-		c,
-		ctx.InformerFactory.Certificates().V1beta1().CertificateSigningRequests(),
-		ctx.ComponentConfig.CSRSigningController.ClusterSigningCertFile,
-		ctx.ComponentConfig.CSRSigningController.ClusterSigningKeyFile,
-		ctx.ComponentConfig.CSRSigningController.ClusterSigningDuration.Duration,
-	)
+	// TODO get different signer cert and key files for each signer when we add flags.
+
+	kubeletServingSigner, err := signer.NewKubeletServingCSRSigningController(c, csrInformer, caFile, caKeyFile, certTTL)
+	if err != nil {
+		return nil, false, fmt.Errorf("failed to start kubernetes.io/kubelet-serving certificate controller: %v", err)
+	}
+	go kubeletServingSigner.Run(1, ctx.Stop)
+
+	kubeletClientSigner, err := signer.NewKubeletClientCSRSigningController(c, csrInformer, caFile, caKeyFile, certTTL)
```

Comment:
Will the change in the way the controllers are being registered cause any change in the way users enable/disable signing controllers?

Reply:
I see a bit of prior discussion now on this 😅 https://github.com/kubernetes/kubernetes/pull/90731#issuecomment-624071680

Reply:
> Will the change in the way the controllers are being registered cause any change in the way users enable/disable signing controllers?

It's got a single big enable block. You cannot choose individual loops here via config.